### DISCUSSION TITLE
Can filters be used to change how the data source is fetched?

### DISCUSSION THREAD

**Discussion by nikspyratos** [Power-Components/livewire-powergrid#1687] [Q&A] [answered]
Heyo. I'm looking for a way to have a table that fetches data from an API, and filters would then correspond to API parameters to change the result set.

Ideally I'd be using a Collection data source, but I don't see any way in the docs or examples on how to pass search or filter data to the data source and completely refresh the source instead of filtering within the table's data set.

Is this possible at all with Powergrid?

**Comment by luanfreitasdev** [comment, marked as answer]
Hello @nikspyratos 

This is not possible in v5.x. But in the next version, this is already planned.

**Reply by nikspyratos** [reply]
Great to hear!

**Reply by nikspyratos** [reply]
Would it alternatively be possible to have an externally-driven data source for a PowerGrid table? Like, feed the collection in via e.g. a wired attribute from blade to the component?

EDIT: The answer is yep, you can just use Livewire reactive properties: https://livewire.laravel.com/docs/nesting#reactive-props

So in your PowerGrid component, add a public property and add `#[Reactive]` to it, and pass it in from the parent's blade.
```
//In your component
#[Reactive]
public $myStuff;

//In your datasource function
return $this->myStuff;

//In blade
<livewire:my-powergrid-table :$myStuff />
```
